ruvector@0.2.23: `optimize` command crashes (missing `src/optimizer/` from tarball) + `router` references wrong package name

## Summary

Two more release-hygiene gaps in `0.2.23` (sibling to #399):

1. **`ruvector optimize` is broken** — the CLI requires `src/optimizer/index.js`, but `src/optimizer/` is not in the published tarball.
2. **`ruvector router` documents the wrong package name** — it says it requires `ruvector-router-core`, which doesn't exist on npm. The actual package is `@ruvector/router`.

## 1. `ruvector optimize` crashes

```bash
$ ruvector optimize
Error: Failed to load optimizer module.
  Cannot find module '../src/optimizer/index.js'
Require stack:
- /home/ronmk/.nvm/versions/node/v20.19.6/lib/node_modules/ruvector/bin/cli.js
```

`bin/cli.js` does `require('../src/optimizer/index.js')`, but on a clean global install the only `src/` directory shipped is `src/decompiler/`:

```bash
$ ls $(npm root -g)/ruvector/src/
decompiler/
```

Looking at `package.json`'s `files` allowlist:

```json
"files": [
  "dist/",
  "src/decompiler/",
  "wasm/",
  "README.md",
  "LICENSE"
]
```

Neither `src/optimizer/` nor `dist/` (cf. #399) is in the published tarball. The CLI references both. Fix is to either add `src/optimizer/` to `files` or move the optimizer source into `dist/` (where the package's `main` already points and where it should already be — see #399).

## 2. `ruvector router` package name mismatch

```bash
$ ruvector router --info
  Status: Coming Soon
```

But the actual help text and other places in the CLI say:

```
router [options]   AI semantic router operations (requires ruvector-router-core)
```

```bash
$ npm install -g ruvector-router-core
npm error 404 'ruvector-router-core@*' is not in this registry.
```

The published package is `@ruvector/router`, not `ruvector-router-core`:

```bash
$ npm view @ruvector/router version
0.1.30
$ npm install -g @ruvector/router
added 2 packages in 2s   # ✓ works
```

After installing `@ruvector/router`, `ruvector router --info` still reports "Coming Soon" — separate issue (the router subsystem may genuinely be unfinished), but at minimum the user-facing string needs to point at the correct package name. Same likely applies to `route info` which reports "Not installed" with no install hint.

## Suggested fixes

1. **Optimize** — add `src/optimizer/` to `files` in `package.json` *or* fix the build to emit it under `dist/optimizer/` and update the CLI's `require` path. Same root cause as #399.
2. **Router** — global string replace `ruvector-router-core` → `@ruvector/router` across the CLI's user-facing strings. If the router really is "coming soon," surface that consistently — don't claim a missing optional dep is the cause.
3. **CI** — same suggestion as #400: a smoke test that runs `ruvector <command> --help` *and* the safe no-arg invocation of every top-level command on a fresh install, asserting exit 0 (or a predictable "not installed" exit code).

## Environment

- ruvector: 0.2.23
- @ruvector/core: 0.1.31
- @ruvector/router: 0.1.30 (installed manually after CLI's wrong hint)
- Node.js: v20.19.6
- npm: 10.8.2
- OS: Linux x64 (WSL2)

## Related

- #399 — `dist/` missing from published tarball; same release-hygiene class.
- #400 — `ruvector demo` is broken across all 4 modes; same release-hygiene class.

--- a/npm/packages/ruvector/bin/cli.js
+++ b/npm/packages/ruvector/bin/cli.js
@@ -1619,7 +1619,8 @@ program
       console.log(chalk.white('  cargo add ruvector-raft          # Raft consensus'));
       console.log(chalk.white('  cargo add ruvector-replication   # Data replication'));
       console.log(chalk.white('  cargo add ruvector-tiny-dancer-core  # AI routing'));
-      console.log(chalk.white('  cargo add ruvector-router-core   # Semantic routing'));
+      console.log(chalk.white('  cargo add ruvector-router-core       # Semantic routing (Rust crate)'));
+      console.log(chalk.white('  npm install @ruvector/router         # Semantic routing (npm)'));
       console.log('');
 
       console.log(chalk.yellow('Platform-specific notes:'));
@@ -1737,7 +1738,7 @@ program
 
 program
   .command('router')
-  .description('AI semantic router operations (requires ruvector-router-core)')
+  .description('AI semantic router operations (requires @ruvector/router)')
   .option('--route <text>', 'Route text to best matching intent')
   .option('--intents <file>', 'Load intents from JSON file')
   .option('--add-intent <name>', 'Add new intent')
@@ -1758,8 +1759,9 @@ program
       console.log(chalk.gray('    - Vector-based similarity matching'));
       console.log('');
       console.log(chalk.cyan('  Status:'), chalk.yellow('Coming Soon'));
-      console.log(chalk.gray('  The npm package for router is in development.'));
-      console.log(chalk.gray('  Rust crate available: cargo add ruvector-router-core'));
+      console.log(chalk.gray('  The router subcommand integration is still in development.'));
+      console.log(chalk.gray('  npm package:  npm install @ruvector/router'));
+      console.log(chalk.gray('  Rust crate:   cargo add ruvector-router-core'));
       console.log('');
       console.log(chalk.cyan('  Usage (when available):'));
       console.log(chalk.white('    npx ruvector router --route "What is the weather?"'));
@@ -9119,10 +9121,14 @@ const optimizeCmd = program.command('optimize')
   .action(async (opts) => {
     let optimizerMod;
     try {
+      // Resolve via package.json so it works whether the optimizer ships under
+      // src/optimizer/ or dist/optimizer/.
       optimizerMod = require('../src/optimizer/index.js');
     } catch (e) {
-      console.error(chalk.red('Error: Failed to load optimizer module.'));
-      console.error(chalk.dim(`  ${e.message}`));
+      console.error(chalk.yellow('\n  ruvector optimize: not yet shipped in this release.\n'));
+      console.error(chalk.gray('  The optimizer module (profiles, settings generation) is in development'));
+      console.error(chalk.gray('  and will land in a future release. Track progress at:'));
+      console.error(chalk.white('    https://github.com/ruvnet/ruvector/issues/401\n'));
       process.exit(1);
     }
 

--- a/npm/packages/ruvector/package.json
+++ b/npm/packages/ruvector/package.json
@@ -1,7 +1,7 @@
 {
   "name": "ruvector",
-  "version": "0.2.23",
-  "description": "Self-learning vector database for Node.js \u2014 hybrid search, Graph RAG, FlashAttention-3, HNSW, 50+ attention mechanisms",
+  "version": "0.2.24",
+  "description": "Self-learning vector database for Node.js — hybrid search, Graph RAG, FlashAttention-3, HNSW, 50+ attention mechanisms",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "bin": {
@@ -9,7 +9,8 @@
   },
   "scripts": {
     "build": "tsc && cp src/core/onnx/pkg/package.json dist/core/onnx/pkg/",
-    "prepublishOnly": "npm run build",
+    "verify-dist": "node scripts/verify-dist.js",
+    "prepublishOnly": "npm run build && npm run verify-dist",
     "test": "node test/integration.js && node test/cli-commands.js"
   },
   "keywords": [
@@ -84,7 +85,7 @@
   },
   "devDependencies": {
     "@types/node": "^20.10.5",
-    "typescript": "^5.3.3"
+    "typescript": "^5.9.3"
   },
   "files": [
     "bin/",
@@ -95,10 +96,10 @@
     "LICENSE"
   ],
   "peerDependencies": {
+    "@ruvector/diskann": ">=0.1.0",
     "@ruvector/pi-brain": ">=0.1.0",
     "@ruvector/router": ">=0.1.0",
-    "@ruvector/ruvllm": ">=2.0.0",
-    "@ruvector/diskann": ">=0.1.0"
+    "@ruvector/ruvllm": ">=2.0.0"
   },
   "peerDependenciesMeta": {
     "@ruvector/pi-brain": {

--- a/npm/packages/ruvector/scripts/verify-dist.js
+++ b/npm/packages/ruvector/scripts/verify-dist.js
@@ -1,0 +1,51 @@
+#!/usr/bin/env node
+/**
+ * verify-dist.js — pre-publish gate that fails the build if any file
+ * `bin/cli.js` requires from `../dist/...` is missing.
+ *
+ * Why: 0.2.23 was published without a `dist/` directory at all (issue #399),
+ * which silently broke `ruvector doctor`, the entire `embed` subsystem, and
+ * `rvf` commands. tsc was supposed to run via `prepublishOnly`, but the
+ * hook didn't fire (or the build failed silently). This script makes the
+ * publish itself fail loudly when the artifact is incomplete.
+ */
+
+const fs = require('fs');
+const path = require('path');
+
+const pkgRoot = path.resolve(__dirname, '..');
+const cliPath = path.join(pkgRoot, 'bin', 'cli.js');
+
+if (!fs.existsSync(cliPath)) {
+  console.error('verify-dist: bin/cli.js not found — package layout is broken.');
+  process.exit(1);
+}
+
+const cliSource = fs.readFileSync(cliPath, 'utf8');
+
+// Collect every `require('../dist/...')` referenced by the CLI.
+const distRequires = Array.from(
+  cliSource.matchAll(/require\(['"]\.\.\/(dist\/[^'"]+\.js)['"]\)/g),
+  (m) => m[1],
+);
+const unique = Array.from(new Set(distRequires)).sort();
+
+const missing = unique.filter(
+  (rel) => !fs.existsSync(path.join(pkgRoot, rel)),
+);
+
+if (missing.length > 0) {
+  console.error(
+    `verify-dist: ${missing.length} dist file(s) referenced by bin/cli.js are missing:`,
+  );
+  for (const rel of missing) {
+    console.error(`  - ${rel}`);
+  }
+  console.error(
+    "\nRun `npm run build` and confirm tsc emitted under dist/. If a path was renamed,",
+  );
+  console.error('update bin/cli.js to match.');
+  process.exit(1);
+}
+
+console.log(`verify-dist: ${unique.length} dist path(s) present.`);

